src/elasticsearch/elasticsearch.ts

```diff
@@ -0,0 +1,52 @@
+import {Instance} from '../aws/types';
+import {ElasticsearchClusterStatus, ElasticsearchNode, Move, MoveCommand, RerouteCommand} from './types';
+import {ssmCommand} from '../utils/ssmCommand';
+import {StandardOutputContent} from 'aws-sdk/clients/ssm';
+import {DefaultResponse} from '../utils/handlerResponses';
+
+export function getClusterHealth(instanceId: string): Promise<ElasticsearchClusterStatus> {
+    return ssmCommand('curl localhost:9200/_cluster/health', instanceId)
+        .then((result: StandardOutputContent) => {
+            const clusterStatus: ElasticsearchClusterStatus = JSON.parse(result);
+            return clusterStatus;
+        })
+}
+
+export function getElasticsearchNode(instance: Instance): Promise<ElasticsearchNode> {
+    console.log(`Searching for Elasticsearch node with private ip: ${instance.privateIp}`);
+    return ssmCommand(`curl localhost:9200/_nodes/${instance.privateIp}`, instance.id)
+        .then((result: StandardOutputContent) => {
+            const json = JSON.parse(result);
+            if (json._nodes.total === 1) {
+                const nodeId: string = Object.keys(json.nodes)[0];
+                return new ElasticsearchNode(instance, nodeId);
+            } else {
+                throw `Expected information about a single node, but got: ${JSON.stringify(json)}`;
+            }
+        })
+}
+
+export function getShardInfo(instanceId: string): Promise<object> {
```

Comment:
seems a shame to return `object` here - depending on how this function is going to be used we could be more specific about the return type?

Reply:
Yes, I agree. Unfortunately, it's a trickier to model this as an interface because there are a number of places where the key names are unknown (see https://github.com/guardian/elasticsearch-node-rotation/pull/11/files#diff-9aa6b149d20be474a85d4702bb61d0d6R25). I spent a little time on this but got stuck and didn't want it to derail progress too much